modules/gapi/src/backends/cpu/gcpuvideo.cpp

```diff
@@ -107,13 +107,59 @@ GAPI_OCV_KERNEL_ST(GCPUBackgroundSubtractor,
     }
 };
 
+GAPI_OCV_KERNEL_ST(GCPUKalmanFilter, cv::gapi::video::GKalmanFilter, cv::KalmanFilter)
+{
+    static void setup(const cv::GMatDesc&, const cv::GOpaqueDesc&,
+                      const cv::GMatDesc&, const cv::gapi::video::KalmanParams& kfParams,
+                      std::shared_ptr<cv::KalmanFilter> &state, const cv::GCompileArgs&)
+    {
+        GAPI_Assert(kfParams.type == CV_32F || kfParams.type == CV_64F);
+        state = std::make_shared<cv::KalmanFilter>(kfParams.dpDim, kfParams.mpDim, kfParams.ctrlDim, kfParams.type);
+
+        // initial state
+        state->statePre = kfParams.statePre;
+        state->errorCovPre = kfParams.errorCovPre;
+
+        // dynamic system initialization
+        state->controlMatrix = kfParams.controlMatrix;
+        state->measurementMatrix = kfParams.measurementMatrix;
+
+        if (cv::norm(kfParams.transitionMatrix, cv::NORM_INF) != 0)
+            state->transitionMatrix = kfParams.transitionMatrix;
+
+        if (cv::norm(kfParams.processNoiseCov, cv::NORM_INF) != 0)
+            state->processNoiseCov = kfParams.processNoiseCov;
+
+        if (cv::norm(kfParams.measurementNoiseCov, cv::NORM_INF) != 0)
+            state->measurementNoiseCov = kfParams.measurementNoiseCov;
+    }
+
+    static void run(const cv::Mat& measurements, bool haveMeasurement,
+                    const cv::Mat& control, const cv::gapi::video::KalmanParams&,
+                    cv::Mat &out, cv::KalmanFilter& state)
+    {
+        cv::Mat pre;
+
+        if (!control.empty() && cv::norm(control, cv::NORM_INF) != 0)
+            pre = state.predict(control);
+        else
+            pre = state.predict();
```

Comment:
Why can't we pass zero-initialized or even empty `control` to `predict()`?
Also, I think `control` can't be passed empty to our kernel

Reply:
If we decide to add overload without control, we will have opportunity to pass empty control vector.

Reply:
I know that Kalman can handle it. But what sense to pass zero control vector which don't change anything in Kalman's state?

Reply:
It makes sense to eliminate `if - else`, just it

Reply:
I want to remind you that control vector is never empty in our design and current implementation. When control vector is zeroed, it is a sign that the controlMatrix is empty. When we pass control vector which is not empty and zeroed to KalmanFilter::predict(), it will try to multiply `control` vector and `controlMatrix`.  But `controlMatrix` is empty. It cause assert.
Before raise proposals please study OpenCV's KalmanFilter class more detailed:
https://github.com/opencv/opencv/blob/b023fcd26418d4da7a73b7ca4abfb21c44be674c/modules/video/src/kalman.cpp#L82

Reply:
OK, I see about empty `comtrolMatrix`, it is true, didn't count that. And also controlMatrix ahve to be release as we can't figure out it's dimensions in case `(ctrlDim > 0)`. Good then! Thanks